packages/eas-cli/src/commands/share.ts

```diff
@@ -0,0 +1,305 @@
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags } = await this.parse(BuildUpload);
+    const { 'build-path': buildPath, fingerprint: manualFingerprintHash } = flags;
+    const {
+      projectId,
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(BuildUpload, {
+      nonInteractive: false,
+    });
+
+    const platform = await this.selectPlatformAsync(flags.platform);
+    const localBuildPath = await resolveLocalBuildPathAsync(platform, buildPath);
+
+    const {
+      fingerprintHash: buildFingerprintHash,
+      developmentClient,
+      simulator,
+    } = await extractAppMetadataAsync(localBuildPath, platform);
+
+    let fingerprint = manualFingerprintHash ?? buildFingerprintHash;
+    if (fingerprint) {
+      if (
+        manualFingerprintHash &&
+        buildFingerprintHash &&
+        manualFingerprintHash !== buildFingerprintHash
+      ) {
+        const selectedAnswer = await promptAsync({
+          name: 'fingerprint',
+          message: `The provided fingerprint hash ${manualFingerprintHash} does not match the fingerprint hash of the build ${buildFingerprintHash}. Which fingerprint do you want to use?`,
+          type: 'select',
+          choices: [
+            { title: manualFingerprintHash, value: manualFingerprintHash },
+            { title: buildFingerprintHash, value: buildFingerprintHash },
+          ],
+        });
+        fingerprint = String(selectedAnswer.fingerprint);
+      }
+
+      await FingerprintMutation.createFingerprintAsync(graphqlClient, projectId, {
+        hash: fingerprint,
+      });
+    }
+
+    Log.log('Uploading your app archive to EAS');
+    const bucketKey = await uploadAppArchiveAsync(graphqlClient, localBuildPath);
+
+    const build = await LocalBuildMutation.createLocalBuildAsync(
+      graphqlClient,
+      projectId,
+      { platform: toAppPlatform(platform), simulator },
+      { type: LocalBuildArchiveSourceType.Gcs, bucketKey },
+      { distribution: DistributionType.Internal, fingerprintHash: fingerprint, developmentClient }
+    );
+
+    Log.withTick(`Here is a sharable link of your build: ${getBuildLogsUrl(build)}`);
+  }
+
+  private async selectPlatformAsync(platform?: Platform): Promise<Platform> {
+    if (platform) {
+      return platform;
+    }
+    const { resolvedPlatform } = await promptAsync({
+      type: 'select',
+      message: 'Select platform',
+      name: 'resolvedPlatform',
+      choices: [
+        { title: 'Android', value: Platform.ANDROID },
+        { title: 'iOS', value: Platform.IOS },
+      ],
+    });
+    return resolvedPlatform;
+  }
+}
+
+async function resolveLocalBuildPathAsync(
+  platform: Platform,
+  inputBuildPath?: string
+): Promise<string> {
+  const applicationArchivePatternOrPath =
+    inputBuildPath ??
+    (platform === Platform.ANDROID
+      ? 'android/app/build/outputs/**/*.{apk,aab}'
+      : 'ios/build/Build/Products/*simulator/*.app');
```

Comment:
I'm not sure if these are compatible with the default artifact path of `eas build --local`. It would create a build and move it to `projectDir/somename.extenstion` and not `projectDir/ios|android/...`. Are these paths the result path as of `npx expo:run` build?

Reply:
yes, I've updated it and now it should match `npx expo:run `